asapdiscovery-data/asapdiscovery/data/schema_v2/ligand.py

```diff
@@ -333,6 +356,102 @@ def set_expansion(
+                    },
+                ),
+            )
+        else:
+            raise ValueError("Unable to generate canonical tautomer")
+
+    @property
+    def has_defined_stereo(self) -> bool:
+        """
+        Check if the ligand has defined stereochemistry
+        """
+        mol = self.to_oemol()
+        return mol.HasPerceived(oechem.OEPerceived_AtomStereo) or mol.HasPerceived(
+            oechem.OEPerceived_BondStereo
+        )
+
+    def is_chemically_equal(self, other: "Ligand") -> bool:
+        """
+        Check if the ligand is chemically equal to another ligand using the inchikey.
+        Both ligands must both have defined stereochemistry or both not have defined stereochemistry.
+        """
+        return (
+            self.fixed_inchikey == other.fixed_inchikey
+            and self.has_defined_stereo == other.has_defined_stereo
+        )
+
+    def is_stereoisomer(self, other: "Ligand"):
+        """
+        Check if the ligand is a possible stereoisomer of another ligand.
+        Returns False if the ligands are the same.
+        """
+        # First check if molecules are the same
+        if self.is_chemically_equal(other):
+            return False
+
+        return self.non_iso_smiles == other.non_iso_smiles
+
+    def is_tautomer(self, other: "Ligand"):
+        """
+        Check if the ligand is a tautomer of another ligand, collapsing protonation states.
+        Returns False if the ligands are the same or stereoisomers.
+        """
+        # First check if molecules are the same or just a stereoisomer
+        if self.is_chemically_equal(other) or self.is_stereoisomer(other):
+            return False
+        return self.canonical_tautomer.is_chemically_equal(other.canonical_tautomer)
+
+    def is_related(self, other: "Ligand"):
+        """
+        Check if the ligand is a steroisomer of a tautomer of another ligand.
+        Returns False if the ligands are the same, stereoisomers, or tautomers with the same stereochem.
+        """
+        # First check if molecules are the same or just a stereoisomer or a tautomer
+        if (
+            self.is_chemically_equal(other)
+            or self.is_stereoisomer(other)
+            or self.is_tautomer(other)
+        ):
+            return False
+        else:
+            # Get the canonical tautomers and check if the non-iso smiles are the same
+            return self.canonical_tautomer.is_stereoisomer(other.canonical_tautomer)
+
+    def get_chemical_relationship(self, other: "Ligand") -> ChemicalRelationship:
+        """
+        Get the chemical relationship between two ligands
+        """
+        if self.is_chemically_equal(other):
+            return ChemicalRelationship.IDENTICAL
+        elif self.is_stereoisomer(other):
+            return ChemicalRelationship.STEREOISOMER
+        elif self.is_tautomer(other):
+            return ChemicalRelationship.TAUTOMER
+        elif self.is_related(other):
```

Comment:
If `is_related` is meant to catch stereo OR tautomer, which is what I am picking up from this flag math, make more explicit with something like `is_stereo_or_tautomer`